iina/OpenURLAccessoryViewController.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  OpenURLAccessoryViewController.swift
+//  iina
+//
+//  Created by lhc on 26/3/2017.
+//  Copyright © 2017 lhc. All rights reserved.
+//
+
+import Cocoa
+
+class OpenURLAccessoryViewController: NSViewController {
+
+  @IBOutlet weak var urlField: ShortcutAvailableTextField!
+
+  @IBOutlet weak var safariLinkBtn: NSButton!
+  @IBOutlet weak var chromeLinkBtn: NSButton!
+  @IBOutlet weak var usernameField: NSTextField!
+  @IBOutlet weak var passwordField: NSSecureTextField!
+
+  private let safariExtensionLink = "https://github.com/lhc70000/iina/releases/download/v0.0.5/Open_In_IINA.safariextz"
+  private let chromeExtensionLink = "https://chrome.google.com/webstore/detail/open-in-iina/pdnojahnhpgmdhjdhgphgdcecehkbhfo"
+
+  var url: URL? {
+    get {
+      let username = usernameField.stringValue
+      let password = passwordField.stringValue
+      guard var urlValue = urlField.stringValue.addingPercentEncoding(withAllowedCharacters: .urlAllowed) else {
+        return nil
+      }
+      if URL(string: urlValue)?.host == nil {
+        urlValue = "http://" + urlValue
+      }
+      guard let urlComponents = NSURLComponents(string: urlValue) else { return nil }
+      if !username.isEmpty && !password.isEmpty {
```

Comment:
Is this necessary? I expect that some websites don't have a password but require a valid username.